tests/modules/lib/cddl-gen/decode/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
```

Comment:
Wouldn't we get away with less maintenance if we merged the tests into a single project. That way we got away with a single cmakelists.txt and a single cddl file.

IMO it is better to have a slightly messy test file and less duplication than a clean test and more complicated and duplicated build system code.

Reply:
The build system code isn't really duplicated, since you need one call to cddl-gen for decode and one for encode. When it's separated like this, the CMakeLists.txt files give an accurate representation of how to add cddl-gen to your project in the different scenarios. Some of the C code is duplicated, but it's not really too much so I think it's ok as is.